crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/mapper/crimeapply/ChildWeightingMapper.java

```diff
@@ -11,62 +11,21 @@ public class ChildWeightingMapper {
-        if(age2to4 > 0) {
-            childWeightings.add(new ChildWeighting(age2to4, ChildWeighting.LowerAgeRange._2, ChildWeighting.UpperAgeRange._4));
-        }
-    }
-
-    private void map5To7AgeChildren(List<Dependant> crimeApplyDependants, List<ChildWeighting> childWeightings) {
-        int age5to7 = Math.toIntExact(crimeApplyDependants.stream().filter(dependant -> (dependant.getAge() >=5 && dependant.getAge() <= 7)).count());
-        if(age5to7 > 0) {
-            childWeightings.add(new ChildWeighting(age5to7, ChildWeighting.LowerAgeRange._5, ChildWeighting.UpperAgeRange._7));
-        }
-    }
-
-    private void map8To10AgeChildren(List<Dependant> crimeApplyDependants, List<ChildWeighting> childWeightings) {
-        int age8to10 = Math.toIntExact(crimeApplyDependants.stream().filter(dependant -> (dependant.getAge() >=8 && dependant.getAge() <= 10)).count());
-        if(age8to10 > 0) {
-            childWeightings.add(new ChildWeighting(age8to10, ChildWeighting.LowerAgeRange._8, ChildWeighting.UpperAgeRange._10));
-        }
-    }
-
-    private void map11To12AgeChildren(List<Dependant> crimeApplyDependants, List<ChildWeighting> childWeightings) {
-        int age11to12 = Math.toIntExact(crimeApplyDependants.stream().filter(dependant -> (dependant.getAge() >=11 && dependant.getAge() <= 12)).count());
-        if(age11to12 > 0) {
-            childWeightings.add(new ChildWeighting(age11to12, ChildWeighting.LowerAgeRange._11, ChildWeighting.UpperAgeRange._12));
-        }
-    }
-
-    private void map13To15AgeChildren(List<Dependant> crimeApplyDependants, List<ChildWeighting> childWeightings) {
-        int age13to15 = Math.toIntExact(crimeApplyDependants.stream().filter(dependant -> (dependant.getAge() >=13 && dependant.getAge() <= 15)).count());
-        if(age13to15 > 0) {
-            childWeightings.add(new ChildWeighting(age13to15, ChildWeighting.LowerAgeRange._13, ChildWeighting.UpperAgeRange._15));
         }
+        return childWeightings;
     }
-    private void map16To18AgeChildren(List<Dependant> crimeApplyDependants, List<ChildWeighting> childWeightings) {
-        int age16to18 = Math.toIntExact(crimeApplyDependants.stream().filter(dependant -> (dependant.getAge() >= 16 && dependant.getAge() <= 18)).count());
-        if (age16to18 > 0) {
-            childWeightings.add(new ChildWeighting(age16to18, ChildWeighting.LowerAgeRange._16, ChildWeighting.UpperAgeRange._18));
+    private void mapChildrenBasedOnAgeRange(List<Dependant> crimeApplyDependants, List<ChildWeighting> childWeightings, ChildWeighting.LowerAgeRange lowerAgeRange, ChildWeighting.UpperAgeRange upperAgeRange) {
+        int numberOfChildren = Math.toIntExact(crimeApplyDependants.stream().filter(dependant -> (dependant.getAge() >= lowerAgeRange.value() && dependant.getAge() <= upperAgeRange.value())).count());
```

Comment:
I didn't notice this before, how many children are we expecting !? lol
`Math.toIntExact`
Agreed, better to safe than sorry! :)